crates/ruma-identifiers/src/mxc_uri.rs

```diff
@@ -14,7 +14,7 @@ use crate::ServerName;
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct MxcUri {
     full_uri: Box<str>,
-    slash_idx: Option<NonZeroU8>,
+    slash_idx: Result<NonZeroU8, MxcUriError>,
```

Comment:
I don't like this being stored at all. I was wondering whether the idx should be stored at all when this type was first introduced, now I feel strongly that it shouldn't, and validation should be done on-the-fly in `.parts()`.

Reply:
It does reduce compute time though, instead of computing every time you call `parts`, it'll call it once upon creation, and then just retrieve the results.

I do agree that this part feels iffy to me though, but I think that making it compute lazily isn't a good change. What do you think?

Reply:
I doubt anybody will ever call `.parts()` nearly often enough for lazyness to ever become an actual problem.

Reply:
Alright, i'll change this then 👍